porch/pkg/engine/engine.go

```diff
@@ -138,6 +138,7 @@ func (cad *cadEngine) CreatePackageRevision(ctx context.Context, repositoryObj *
 		if err != nil {
 			return nil, err
 		}
+		mutations = append(mutations, &addMergeKeyMutation{})
```

Comment:
Why do this in a separate mutation rather than as part of the update? Since this is essentially an implementation detail of the merge logic, I don't think we need to make this a separate mutation (which leads to tasks and separate commits).